applications/asset_tracker_v2/doc/location_module.rst

```diff
@@ -0,0 +1,113 @@
+
+Features
+********
+
+This section documents the various features implemented by the module.
+
+Location control
+================
+
+The module uses the :ref:`lib_location` to communicate with the nRF9160 modem and control its GNSS and LTE neighbor cell measurement functionalities.
+A location request starts when the module receives an ``APP_EVT_DATA_GET`` event and the ``APP_DATA_LOCATION`` type is listed in the event's ``data_list`` member containing the data types that shall be sampled.
+
+The default configuration for the location request is set as follows:
+
+* GNSS set as the first priority method and cellular positioning as the second priority.
+  Methods that should not be used can be configured by adding them into ``No Data List``.
+  See :ref:`Real-time configurations <real_time_configs>` for more information.
+* Timeout which can be configured as part of :ref:`Real-time configurations <real_time_configs>`.
+
+The location module receives configuration updates as payload in the following two events:
+
+* ``DATA_EVT_CONFIG_INIT`` - The event contains the value configured at build time or received from cloud and stored to flash.
+  The event is distributed from the data module as part of the application initialization.
+* ``DATA_EVT_CONFIG_READY`` - The event is received when the data module has processed an incoming message from cloud with a configuration update.
+
+The module sends the following events based on the outcome of the location request:
+* ``LOCATION_MODULE_EVT_GNSS_DATA_READY``; GNSS position has been acquired
+* ``LOCATION_MODULE_EVT_NEIGHBOR_CELLS_DATA_READY``: neighbor cell measurements are ready to be sent to cloud for location resolution
+* ``LOCATION_MODULE_EVT_DATA_NOT_READY``: Location request failed
+* ``LOCATION_MODULE_EVT_TIMEOUT``: Timeout occurred
+* ``LOCATION_MODULE_EVT_AGPS_NEEDED``: A-GPS request should be sent to cloud
+* ``LOCATION_MODULE_EVT_PGPS_NEEDED``: P-GPS request should be sent to cloud
+
+GNSS LNA configuration
+======================
+
+Different devices have different GNSS antenna and LNA setups depending on the antenna type (onboard or external).
+The application uses the :ref:`lib_modem_antenna` library for configuring the LNA.
+The library has Kconfig options to control the LNA using either the COEX0 or MAGPIO interface of the nRF9160.
+See the library documentation for more details on how to configure the antenna and LNA.
+
+GPS assistance data
+===================
+
+The location module receives requests for GPS assistance data from the :ref:`lib_location` library.
+When the module receives an A-GPS request, it distributes it to the other modules as a ``LOCATION_MODULE_EVT_AGPS_NEEDED`` event that contains information about the type of assistance data needed.
+Providing the requested A-GPS data may reduce the time it takes to acquire a GNSS fix.
+
+Module internals
+================
+
+The location module is a threadless module in terms of message processing.
```

Comment:
Good that you found this. I was supposed to ask whether we should have own thread for the location module message processing.
For GNSS module, we had a thread for processing GNSS events from the modem library.
For cloud, data, debug, modem and sensor modules we have a thread for processing their messages. Rest of the modules don't have own thread.

Reply:
@simensrostad Any comments on this?

Reply:
As long as none of the location API calls blocks for a prolonged amount of time there shouldn't be any need to have thread.

Reply:
Location API calls are rather quick because the library has so long lasting functionality that any time consuming information is coming later in the event callback. I'll modify this text.